doc/company.texi

```diff
@@ -1305,22 +1309,136 @@ switching to @emph{company-capf} backend.
 @subsection Natural Language Completion
```

Comment:
Perhaps we should call this "Text Completion" or "Prose Completion"?

I'm not 100% sure, but I think the phrase "natural language" comes up more often in specialized contexts, where one also deals with the (vague, complex, but still existing) rules of natural languages and the meanings behind words. And we just provide dictionary completion, of sorts.

Which (Dictionary Completion) could be another option for the name of this subsection.

Reply:
"Text Completion" seems to be a good enough title, and a simple one. Updated to use it.
Note that the titles are used in the generated URLs, so it might help avoid issues in the future to choose the "durable".  So let me know if you see a vague title.

Reply:
Regarding other titles:

"Templates Expansion" could be renamed to "Snippet Expansion". These are more or less equivalent, but in my mind (at least) a template usually has some place-holders for values. And `company-abbrev` (or `abbrev` package in general) does not do that.

"File Paths Completion" -- according to GNU standards, a "path" is a list of values (like the value of an env var `PATH` on unixy systems). Whereas an absolute or relative file name (with directory parts, etc) is still a file name. So the section should probably be called "File Name Completion".

Reply:
Thank you for looking into this.

> "Templates Expansion" could be renamed to "Snippet Expansion".  These are more or less equivalent

I agree with you, these words seem to be more or less interchangeable and "snippet" might be a bit more suitable choice overall. 

After looking into updating the text, it appeared there's a reason to stick with the "template": both `Tempo` and `YASsnippet` define themselves with this term. Respectively: "Flexible template insertion" and ["is a template system"](https://github.com/joaotavora/yasnippet). Changing the current text feels clumsy because of this, so let's keep the current version.

[BTW, (probably you're aware of it) `abbrev` allows to configure a hook function for rudimentary additional actions (f.i. move point to a place-holder in the expansion).]

> File Name Completion

It's an interesting catch. From a targeted reader perspective, it may confuse (since the term "path" is common in this context), but the section's text should clarify what exactly is to be completed.  

Would it be fine to use the plural form "File Names Completion" then? Otherwise, maybe turn siblings into the singular form: "Symbol Name Completion", "Template Expansion"?

I was also going to suggest POSIX's "Pathname" term, but then stumbled upon this text in `(emacs) Glossary`, `File Name` section:

> Some people use the term `pathname` for file names, but we do not;
we use the word `path` only in the term `search path` (q.v.).

Reply:
> After looking into updating the text, it appeared there's a reason to stick with the "template": both Tempo and YASsnippet define themselves with this term. Respectively: "Flexible template insertion" and "is a template system". Changing the current text feels clumsy because of this, so let's keep the current version.

But it is ya*snippet*, right? And tempo is probably old enough that its description was written back when the term "snippet" didn't enter the common language.

And if we look at VS Code's docs (which most likely reflect the popular usage), the main name is "snippet", which is then clarified to be a [kind of] template: https://code.visualstudio.com/docs/editor/userdefinedsnippets

Anyway, you to you, it's not too serious an issue either way.

> [BTW, (probably you're aware of it) abbrev allows to configure a hook function for rudimentary additional actions (f.i. move point to a place-holder in the expansion).]

Not at all. I've never used this backend either. Good to know, though. :-)

> Would it be fine to use the plural form "File Names Completion" then? Otherwise, maybe turn siblings into the singular form: "Symbol Name Completion", "Template Expansion"?

I think the singular form is best. File Name Completion, Symbol Name Completion, etc.

We could also replace "Symbol Name Completion" with "Code Completion". Which is a more general term, and it's what `company-capf` does, really. Especially when used together with "smart" completion providers, such as, for example, the LSP clients.

> I was also going to suggest POSIX's "Pathname" term, but then stumbled upon this text in (emacs) Glossary, File Name section:

Yup.

Reply:
> the main name is "snippet", which is then clarified to be a [kind of] template

Funny enough, this could be looked at as one more voice for the "template" term: since it defines a superset of "snippet", and a more general title is preferred.

> it's not too serious an issue either way.

+1

> I think the singular form is best.  ... We could also replace "Symbol Name Completion" with "Code Completion". 

Thanks! The names looks much cleaner now.

Reply:
> since it defines a superset of "snippet", and a more general title is preferred

But we don't provide expansion of any other kind of templates?

Reply:
> But we don't provide expansion of any other kind of templates?

Yes, sure. I meant to state "just in case, for future possible (even if unlikely) changes".